src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.cs

```diff
@@ -497,7 +497,7 @@ private void Free(HashSet<CompilationEvent> events)
         /// </summary>
         public bool HasPendingSyntaxAnalysis(AnalysisScope analysisScope)
         {
-            if (analysisScope.IsTreeAnalysis && !analysisScope.IsSyntaxOnlyTreeAnalysis)
+            if (analysisScope is { IsTreeAnalysis: true, IsSyntaxOnlyTreeAnalysis: false })
```

Comment:
>is [](start = 30, length = 2)

nice

Reply:
@alrz Action item is to make sure a test captures this as a desired case for the refactoring to support.